packages/grafana-ui/src/components/Table/Table.tsx

```diff
@@ -282,14 +282,15 @@ export class Table extends Component<Props, State> {
       this.scrollToTop = false;
     }
 
+    // Force MultiGrid to rerender if these options change
+    const refreshKeys = {
+      d0: data,
+      d1: this.props.data,
+      s0: this.props.styles,
+    };
     return (
       <MultiGrid
-        {
-          ...this.state /** Force MultiGrid to update when data changes */
-        }
-        {
-          ...this.props /** Force MultiGrid to update when data changes */
-        }
+        {...refreshKeys}
```

Comment:
Not sure about this approach as it's sort of hacky to push props to a component that are not handled by it. Shouldn't we use `key` instead that would be an outcome of the changed properties?  You could generate it in `getDerivedStateFromProps` and pass it to `MultiGrid` in render.

Reply:
I can't find it in the docs right now... but this was their recommendation on how do get things to update when the data changes.  `key` alone does not work because MultiGrid includes 4 Grids that don't have any context to know what changed besides the `...rest` parameters.  

This whole file needs to be revisited.  It was my first attempt at anything non-trivial in react *and* blindly copied over a lot of angular code :(

Reply:
Check: 
https://github.com/bvaughn/react-virtualized#pure-components

Essentially to get MultiGrid to update the inner Grids, you either add pass-through props, or call `forceUpdateGrids` directly

Reply:
Cool, I get it, thanks!